### DISCUSSION TITLE
[Submarine] Minor Adjustments

### DISCUSSION THREAD

**Discussion by Koku0** [FakeFishGames/Barotrauma#11792] [Ideas and suggestions]
### Minor Problems
- Certain submarines that can never mount railguns have railgun shell racks.
- In an overview, such as selecting the submarine for a multiplayer lobby, some connected hulls have different names to each other while some unconnected hulls have the same names, which could be clarified.
- In some subs, ballast tanks are labelled "A" and "B" on the status monitor, but the rooms above the ballasts have no corresponding signage. Otherwise, they could be renamed to Front and Rear Ballasts.

Granted, these problems could be intentional design choices but I simply don't know. For each sub adjusted, an image will be attached. For non-visible changes, I've written the change into the list of dot-points.

The goal of these changes is to make navigating subs and following a Captain's orders easier and less confusing.

### What Submarines Need These Adjustments?
**Dugong**
- Has a railgun shell rack, but cannot mount a railgun. It has been replaced with an additional ammunition shelf.
- Has two Upper-Mid hulls. The left hull renamed to Airlock Access, as that is more accurate to its function.
- Ballasts A & B are unlabelled and have been renamed to Rear Ballast and Front Ballast, respectively.
![Dugong](https://user-images.githubusercontent.com/47022005/230819134-e489acc8-3db8-47f0-b8d3-065a561c6a23.png)

**Orca**
- Has a railgun shell rack, but cannot mount a railgun. It has been replaced with an additional ammunition shelf.
- Ballasts A, B, & C are unlabelled and have been labelled.
- Differentiated both Docking Ports to Upper and Lower to reduce confusion.
![Orca](https://user-images.githubusercontent.com/47022005/230819150-db3a1c76-23f7-4178-9e6d-b66893346b14.png)

**Humpback**
- Some hulls slightly overlapped with one another, which has since been fixed.
(No image because changes are invisible)

**Orca2**
- Docking Port renamed to Airlock Access and the right Airlock hull renamed to Docking Port, to be more accurate to the function of the rooms. Both retain the "Is Wet Room" and "Avoid Staying" properties.
- Engineering hull with the O2 generator renamed to Oxygen Compartment.
- Upper Mid hull renamed to Mid Aft Side.
(No image because changes are invisible)

**Azimuth**
Discussed here: #11675 

**Typhon**
Discussed here: #11783 

**Herja**
- Ballasts A & B are unlabelled and have been renamed to Rear Ballast and Front Ballast, respectively.
(No image because changes are invisible)

**Berilia**
- Bilge room above Ballast renamed to Lower Mid, as the Berilia contains many Bilge rooms. It retains the "Is Wet Room" and "Avoid Staying" properties.
(No image because changes are invisible)

**Remora**
- Remora Drone's Command Room renamed to Drone. Drone's Ballast renamed to Drone Ballast. This is to reduce confusion with the Remora's own Command and Ballast rooms.
- Remora's drone control room renamed to Drone Control. The room to the left of the Command Room with the navigation terminal has been renamed to Armory. This may prevent confusion, owing to the sub's sprawling and decentralised layout.
- Both Forward Access Shafts have been linked, as there is no hatch separating them.
(No image because changes are invisible)

**Typhon2**
- Top Airlock Access has been renamed to Docking Port to reduce confusion between both Airlock Accesses.
(No image because changes are invisible)

**Kastrull**
- Renamed the lower Airlock Access to Lower Docking Port, to prevent confusion with the upper Airlock Access. It retains the "Is Wet Room" and "Avoid Staying" properties.
- Ballasts A, B, & C are unlabelled and have been labelled.
![Kastrull](https://user-images.githubusercontent.com/47022005/230819191-ff3cc1c4-e105-4ebe-95b0-b70c048c44f5.png)

**Winterhalter**
- Renamed Stowage to Armory, as that better fits its purpose.
- I didn't change anything, but I'm not sure how fitting it is for a Scout-class submarine to have two railgun shell racks when Attack-class subs of the same Tier only have one.
(No image because changes are invisible)

### Conclusion
I hope that I've been clear with the changes I've made and that they're understandable. I know these are minor issues, but I hope fixing them will make navigating these ships slightly easier and improve the quality of life on certain vessels. Attached are the adjusted .sub files in a ZIP folder, if you want them.
[Vanilla Sub Reworks.zip](https://github.com/Regalis11/Barotrauma/files/11187215/Vanilla.Sub.Reworks.zip)

Any feedback is much appreciated :)

**Comment by ubertpendragon** [comment]
You probably should've opened this as an issue, I was about to post about those vanilla sub problems

Your Kastrull is outdated though, the unstable version has a layout change (door leading to armory)

**Reply by Koku0** [reply]
Ah ok, I thought to put these here because they're not major issues. I'll open an issue for this and delete this post.

**Comment by ubertpendragon** [comment]
I do remember that railgun racks on subs that can't use railguns were kept for looting and modding reasons 

But yes the hull renaming makes sense

**Reply by Koku0** [reply]
The railgun racks make sense in that regard, but it feels like if you have looted shells you're not going to keep them if you can't use them, so why would precious space be devoted to storing them? For modding reasons, if someone is going to give the Dugong a railgun, surely they have the ability to give it a railgun rack as well?

The reasons you've stated make sense, don't get me wrong, but it doesn't feel like sufficient justification to put something that isn't going to be useful ever on a sub.

**Reply by Ek-Videogames** [reply]
With how heavily sell price & deconstruction got nerfed, it's not even worth the effort of lugging the rail ammo back because there's nothing to do with it - unless you have a railgun to use it, or know you're about to switch to a sub that has a railgun.
